### PR TITLE
feat: added bps and bps_xz

### PR DESCRIPTION
**Pull request by joshieman06**
xz does not use change in y.

### CODE REVIEW

File: common/src/main/java/com/wynntils/functions/CharacterFunctions.java [hunk 1]
```diff
@@ -37,6 +42,28 @@ public List<String> getAliases() {
         }
     }
 
+    public static class bpsFunction extends Function<Float> {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/functions/CharacterFunctions.java]: Always use UpperCamel for classes

File: common/src/main/java/com/wynntils/functions/CharacterFunctions.java [hunk 2]
```diff
@@ -37,6 +42,28 @@ public List<String> getAliases() {
...
+        public Float getValue(String argument) {
+            LocalPlayer player = player();
+            double dX =  player.getX() - player.xOld;
+            double dZ =  player.getZ() - player.zOld;
+            double dY =  player.getY() - player.yOld;
+            return (float) Math.sqrt((dX*dX)+(dZ*dZ)+(dY*dY))*20;
+            }
+    }
+
+    public static class bpsXzFunction extends Function<Float> {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/functions/CharacterFunctions.java]: UpperCamel.

File: common/src/main/java/com/wynntils/functions/CharacterFunctions.java [hunk 3]
```diff
@@ -7,11 +7,16 @@
...
 import com.wynntils.core.managers.Model;
+import com.wynntils.mc.utils.McUtils;
 import com.wynntils.utils.StringUtils;
 import com.wynntils.wynn.model.ActionBarModel;
 import com.wynntils.wynn.model.CharacterManager;
+import net.minecraft.client.player.LocalPlayer;
+
 import java.util.List;
 
+import static com.wynntils.mc.utils.McUtils.player;
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/functions/CharacterFunctions.java]: Remove static imports. I do not know if spotless complains, but I sure do :)

File: common/src/main/java/com/wynntils/core/functions/FunctionManager.java [hunk 4]
```diff
@@ -336,5 +338,6 @@ public static void init() {
         registerFunction(new MinecraftFunctions.ZFunction());
         registerFunction(new MinecraftFunctions.DirFunction());
         registerFunction(new MinecraftFunctions.FpsFunction());
+
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/core/functions/FunctionManager.java]: I dont know why you added this

**Review by kristofbolyai — COMMENTED** [review]
Few touchups and looks good!

File: common/src/main/java/com/wynntils/core/functions/FunctionManager.java (hunk 4 above)
Comment (joshieman06) [reply common/src/main/java/com/wynntils/core/functions/FunctionManager.java]: me either

File: common/src/main/resources/assets/wynntils/lang/en_us.json [hunk 5]
```diff
@@ -608,6 +608,10 @@
   "feature.wynntils.wynntilsQuestBook.name": "Wynntils Quest Book",
   "feature.wynntils.wynntilsQuestBook.replaceWynncraftQuestBook.description": "Should the Wynncraft quest book be replaced with the mod's custom one?",
   "feature.wynntils.wynntilsQuestBook.replaceWynncraftQuestBook.name": "Replace Wynncraft Quest Book",
+  "function.wynntils.bps.description": "Player speed in blocks per second",
+  "function.wynntils.bps.name": "Player Speed",
+  "function.wynntils.bpsXz.description": "Player speed in blocks per second without the change in y",
```
Comment (kristofbolyai) [inline common/src/main/resources/assets/wynntils/lang/en_us.json]:
```suggestion
  "function.wynntils.bpsXz.description": "Player speed in blocks per second without the change in the y coordinates",
```

**Comment by joshieman06** [comment]
fixing format atm

**Review by kristofbolyai — APPROVED** [review]

**Pull request merged**